akka-persistence/src/test/scala/akka/persistence/PersistentActorStashingSpec.scala

```diff
@@ -163,8 +163,12 @@ class SteppingInMemPersistentActorStashingSpec extends PersistenceSpec(
       SteppingInmemJournal.step(journal)
       SteppingInmemJournal.step(journal)
 
-      persistentActor ! GetState
-      expectMsg(List("a", "c", "b"))
+      within(3.seconds) {
+        awaitAssert {
+          persistentActor ! GetState
+          expectMsg(List("a", "c", "b"))
+        }
+      }
```

Comment:
The problem could also be solved by injecting the an "replyTo" (or rather `stateProbe`) test probe into the persistentActor, such that it informs the probe on applying `updateState` about the current one, then the race between step-induced processing and `GetState` wouldn't happen.

I don't really mind this solution, it's just a test and bound to complete successfully quickly IMO. So it can stay as is.